File: vmlib/src/main/java/me/shouheng/vmlib/anno/ServiceConfiguration.java
package me.shouheng.vmlib.anno;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Configuration annotation for Android Service.
 *
 * @author <a href="mailto:shouheng2015@gmail.com">WngShhng</a>
 * @version 2019-7-1
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ServiceConfiguration {

    /**
     * Will the event bus will be used in this view.
     *
     * @return true if you want to use event bus.
     */
    boolean useEventBus() default false;
}

File: vmlib/src/main/java/me/shouheng/vmlib/anno/ServiceConfiguration.kt
package me.shouheng.vmlib.anno

/**
 * Configuration annotation for Android Service.
 *
 * @author [WngShhng](mailto:shouheng2020@gmail.com)
 * @version 2019-7-1
 */
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class ServiceConfiguration(

    /** Will the event bus will be used in this view. */
    val useEventBus: Boolean = false
)